pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2613,6 +2613,15 @@ type StorageScopeSpec struct {
 	StorageClassDeviceSets []StorageClassDeviceSet `json:"storageClassDeviceSets,omitempty"`
 	// +optional
 	Store OSDStore `json:"store,omitempty"`
+	// +optional
+	// RestartInterval defines the time for which the OSD pods, that failed with zero exit code, will sleep before restarting.
+	// This is useful in scenarios like OSD flapping, when OSDs are marked down more than 5 times in 600 seconds by Ceph.
+	// Preventing the OSD pods to restart immediately in such scenarios will prevent Rook from marking OSD as `up` and thus
+	// peering of the PGs mapped to the OSD.
+	// The interval defaults to 24 hours if no value is provided. User needs to manually restart the OSD pod if they manage to fix
+	// the underlying OSD flapping issue before the restart interval.
+	// Provide time unit in hours. For example, 2h, 6h, etc.
+	RestartInterval string `json:"restartInterval"`
```

Comment:
How about if we just name it like this and use an int? Then they don't need to worry about units in the string.
```suggestion
	FlappingRestartIntervalHours int `json:"restartInterval"`
```